libraries/Standard/src/math/special.birch

```diff
@@ -363,6 +366,10 @@ function upper_inc_gamma(a:Real, x:Real) -> Real {
 
   assert x >= Real(0.0);
```

Comment:
Still quite a few of these casts of literals around, might just want to do a search and replace to remove them :)

Reply:
Oh, maybe I misunderstood, I thought we wanted to cast to Real explicitly for the float64 implementations and similarly for the float32 implementations to make this GPU compatible?

Reply:
I may not have explained very well! Let me try again:

1. There are types `Real64` and `Real32`.
2. `Real` is just an alias for `Real64` (there's a `type Real = Real64` in the standard library).
3. Literals like `0.0` are of type `Real`.
4. When we start wanting to use 32-bit (i.e. single precision) floating point, there are a couple of things we may want to do. One is to provide explicit `Real64` and `Real32` overloads of relevant functions, as for the standard math functions already (e.g. `exp(x:Real64)` and `exp(x:Real32)`). For the 32-bit overloads we will need to cast literals to `Real32` where necessary, to ensure that we are not unintentionally promoting 32-bit floating point computations to 64-bit (in Birch, as in C/C++, floats are implicitly promoted to doubles when in expressions with them). An alternative is to enable 32-bit literals in Birch, which in C/C++ are written e.g. `0.0f`.
5. The other thing we may want to do is what [LibBi](https://libbi.org) already does: provide a build option like `--enable-single` that redefines `type Real = Real32`, thus making all use of `Real` now mean 32-bit, including literals being 32-bit.

So `Real32(0.0)` is and will be a meaningful thing to do to ensure that operations are 32-bit, but `Real(0.0)` doesn't change anything---the literal `0.0` is already of type `Real`, whether `Real = Real64` or `Real = Real32`.

Reply:
Fixed :)